File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // sort(nums.begin(),nums.end());
        // int n=nums.size();
        // return nums[n/2];

        //2nd approach

        // int cnt=0;
        // int el;
        int n=nums.size();
        // for(int i=0;i<n;i++)
        // {
        //     if(cnt==0)
        //     {
        //         cnt=1;
        //         el=nums[i];
        //     }
        //     else if(el==nums[i])cnt++;
        //     else cnt--;
        // }
        // int cnt1=0;
        // for(int  i=0;i<n;i++)
        // {
        //     if(nums[i]==el) cnt1++;
        // }
        // if(cnt1>(n/2)) return el;
        // return -1;

        // 3rd approach
         unordered_map<int,int>mpp;
         for(int i=0;i<n;i++)
         {
            mpp[nums[i]]++;
         }

         for(auto x: mpp)
         {
             if(x.second>n/2)
               return x.first;
         }
             return -1;
     }
};

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
   let e=0;
   let c=0;

   for(let i=0;i<nums.length;i++)
   {
       if(c==0) 
       {
           e=nums[i];
       }
       if(e == nums[i]){
           c++;
       }
       else
       {
           c--;
       }
   }
   return e;

};